apps/admin/backend/src/store.ts

```diff
@@ -1274,6 +1325,149 @@ export class Store {
     }
   }
 
+  deleteAllManualTallies({ electionId }: { electionId: Id }): void {
+    this.client.run(
+      `delete from manual_tallies where election_id = ?`,
+      electionId
+    );
+
+    // removing manual tallies may have left unofficial write-in candidates
+    // without any references, which we clean up
+    this.deleteAllChildlessWriteInCandidates();
+  }
+
+  setManualTally({
+    electionId,
+    precinctId,
+    manualTally,
+  }: {
+    electionId: Id;
+    precinctId: PrecinctId;
+    manualTally: ManualTally;
+  }): void {
+    const ballotCount = manualTally.numberOfBallotsCounted;
+    const serializedContestTallies = JSON.stringify(manualTally.contestTallies);
+
+    const { id: manualTallyRecordId } = this.client.one(
+      `
+        insert into manual_tallies (
+          election_id,
+          precinct_id,
+          ballot_count,
+          contest_tallies
+        ) values 
+          (?, ?, ?, ?)
+        on conflict
+          (election_id, precinct_id)
+        do update set
+          ballot_count = excluded.ballot_count,
+          contest_tallies = excluded.contest_tallies
+        returning (id)
```

Comment:
i've never seen us use `on conflict` or `returning` in our codebase but it allows some efficiencies - fewer database requests - and i think it's fair game